apps/src/util/dropletCommon.js

```diff
@@ -0,0 +1,76 @@
+/*globals dashboard*/
+import trackEvent from '@cdo/apps/util/trackEvent';
+
+/**
+ * Runs the user's droplet code through the droplet parser. If droplet is unable
+ * to parse the user's code, call the error callback with any information we
+ * have regarding the error. Additionally, this will return `true` if any
+ * errors are detected and `false` otherwise.
+ * @param {object} dropletEditor The droplet engine
+ * @param {function} errorCallback The function to call if an error is detected.
+ *                   Usage: errorCallback(lineNumber, errorMessage)
+ * @returns {boolean} true if an error was detected. false otherwise.
+ */
+const findDropletParseErrors = (dropletEditor, errorCallback) => {
+  if (!dropletEditor) {
+    return false;
+  }
+
+  try {
+    dropletEditor.parse();
+  } catch (error) {
+    let errorObject;
+    try {
+      errorObject = JSON.parse(error.message);
+    } catch (_) {
+      // Do nothing. We are just checking if the error is in a JSON format.
```

Comment:
What we should have done here is logged information about the stack trace to GA or to firehose then `return false`. Even though there is a droplet error in the student's code, we don't have enough information about the error to cause their code to fail. 

However, we should allow this behavior to be overridden by the calling function. This will be needed when a student is trying to transition from code mode to droplet mode. That transition will fail and we need to pass any information we have to the user. So they can try to figure out why their code can't transition to droplet mode.

Reply:
One other thought: (This isn't a necessary change, but...) we should probably log the details to firehose rather than Google Analytics. Given the uniqueness of the errors, Firehose would do a better job of letting us collect errors into groups.